### PR TITLE
new command line argument --cert-passphrase added.

### PR DESCRIPTION
**Pull request by mirosyn**
For decrypting the private key that is provided with the --certs command

#### Description

New command line argument added "--cert-passphrase", OpenSSL.crypto.load_privatekey already supports the passphrase, and it was only missing the arguments to pass the passphrase.

#### Checklist

 - [x] I have updated tests where applicable.
 - [x] I have added an entry to the CHANGELOG.

### CODE REVIEW

File: test/mitmproxy/test_proxy.py [hunk 1]
```diff
@@ -42,11 +42,12 @@ def test_simple(self):
...
-            "--cert",
-            tdata.path("mitmproxy/data/testkey.pem"))
-        with pytest.raises(Exception, match="does not exist"):
-            self.p("--cert", "nonexistent")
+        with pytest.raises(Exception, match="ambiguous option"):
+            self.assert_noerr(
+                "--cert",
+                tdata.path("mitmproxy/data/testkey.pem"))
+            with pytest.raises(Exception, match="does not exist"):
+                self.p("--cert", "nonexistent")
```
Comment (mhils) [inline test/mitmproxy/test_proxy.py:50]: Is this change intentional? The double-indented `.raises` doesn't make much sense to me.

File: mitmproxy/certs.py [hunk 2]
```diff
@@ -196,7 +196,7 @@ def load_dhparam(path):
             return dh
 
     @classmethod
-    def from_store(cls, path, basename, key_size):
+    def from_store(cls, path, basename, key_size, passphrase: str = None):
```
Comment (mhils) [inline mitmproxy/certs.py:199]: Let's make this argument bytes right away and call `options.cert_passphrase.encode()` when calling it. The earlier we convert this, the more likely we are to not run into encoding issues. This also removes the slightly ugly `None if passphrase is None else ` :)

```suggestion
    def from_store(cls, path, basename, key_size, passphrase: typing.Optional[bytes] = None):
```

File: mitmproxy/options.py [hunk 3]
```diff
@@ -47,6 +47,10 @@ def __init__(self, **kwargs) -> None:
             certificate as the first entry.
             """
         )
+        self.add_option(
+            "cert_passphrase", Optional[str], None,
+            "Passphrase for decyrpting the private key provided in the --cert setting."
```
Comment (mhils) [inline mitmproxy/options.py:52]:
```suggestion
            "Passphrase for decrypting the private key provided in the --cert option."
```

**Review by mhils — COMMENTED** [review]
This looks like a solid PR overall, thanks! 🍰 😃

File: test/mitmproxy/test_proxy.py (hunk 1 above)
Comment (mirosyn) [reply test/mitmproxy/test_proxy.py]: Indeed double .raises doesn't make sense, removed the original, since the exception is now different, "ambiguous option" --certs and --cert-passphrase resembles each others and it now complains about that.

File: mitmproxy/certs.py (hunk 2 above)
Comment (mirosyn) [reply mitmproxy/certs.py]: Changed it, but we still need one "ugly" if statement in config.py

File: mitmproxy/options.py (hunk 3 above)
Comment (mirosyn) [reply mitmproxy/options.py]: Done

**Comment by mirosyn** [comment]
ping @mhils . did some changes based on the review comments.

**Comment by mhils** [comment]
Awesome, thanks! 🍰
